src/components/Profile/index.tsx

```diff
@@ -76,6 +77,12 @@ const ViewProfile: NextPage = () => {
           <Details profile={profile as any} />
         </GridItemFour>
         <GridItemEight className="space-y-5">
+          {currentProfile && profile && currentProfile.id !== profile.id && (
```

Comment:
Why should we verify `currentProfile && profile` ? 
```suggestion
          {currentProfile?.id !== profile?.id && (
```
can we do it this way?

Reply:
@harish-sethuraman if you keep only statement `currentProfile?.id !== profile?.id`: if `currentProfile == undefined` and `profile == undefined` then this statement will become `true` which is the case I was trying to avoid.

But I see there's `if (!data?.profile) { 404 }` so we can assume that at least `profile` is not empty, so we can change the statement like you suggested.